docs/en/03_Upgrading/02_Upgrading_module.md

```diff
@@ -0,0 +1,335 @@
+-    "type": "silverstripe-module",
++    "type": "silverstripe-vendormodule",
+    "require": {
+-        "silverstripe/framework": "^3"
++        "silverstripe/framework": "^3",
++        "silverstripe/vendor-plugin": "^1"
+    }
+}
+```
+
+### Prefer specific modules over recipes
+
+When upgrading a project, it is recommended to require recipes over modules. However, when upgrading a module, you want to limit the number the number of additional packages that gets installed along with your module. You should target specific packages that your module dependents on. 
+
+For example, let's say your module adds a new ModelAdmin to the SilverStripe administration area without interacting with the CMS directly. In this scenario the main module you will need is `silverstripe/admin` which contains the `ModelAdmin` class and related administration functionality. If you update your `composer.json` file to require `silverstripe/recipe-cms`, you'll force your users to install a lot of modules they may not need like `silverstripe/cms`, `silverstripe/campaign-admin`, `silverstripe/asset-admin`, `silverstripe/versioned-admin`.
+
+### Avoid rigid constraints
+
+Choose constraint based on the minimum version of SilverStripe 4 you are planning on supporting and allow your module to work with future releases. 
+
+For example, if your module requires an API that got introduced with the 4.1 release of `silverstripe/framework`, then that's the version you should target. You should use the caret symbol (`^`) over the ellipse (`~`) so your module works with more recent releases. In this scenario, your constraint should look like `"silverstripe/framework": "^4.1"`.
+
+### Avoid tracking unnecessary files
+
+If you run composer commands from your module's folder, a lock file will be created and dependencies will be installed in a vendor folder. You may also get `project-files` and `public-files` entries added under the `extra` key in your composer.json.
+
+While these changes may be useful for testing, they should not be part of the final release of your module.
+
+### Finalising the module's dependency upgrade
+
+You should commit the changes to your module's `composer.json` and push them to your remote branch.
+
+By this point, your module should be installable in a test SilverStripe 4 project. It will be installed under the vendor directory (e.g.: `vendor/example-user/silverstripe-example-module`). However, it produce exception if you try to run it.
+
+From this point, you can either work from a test project or you can keep working directly on your module.
+
+## Step 2 - Update your environment configuration
+
+As a module maintainer, you shouldn't be shipping any environment file with your module. So there's no need for you to run the upgrader `environment` command. If your module requires any environment variables, you should update your documentation accordingly, but otherwise you can move on to the next step.
```

Comment:
```suggestion
As a module maintainer, you shouldn't be shipping any environment file with your module. So there's no need for you to run the upgrader `environment` command. If your module requires environment variables, you should update your documentation accordingly, but otherwise you can move on to the next step.
```